compiler/ras/DebugCounter.cpp

```diff
@@ -375,10 +478,36 @@ TR::DebugCounter *TR::DebugCounterGroup::findCounter(const char *nameChars, int3
    return _countersHashTable[hi];
    }
 
-TR::DebugCounterAggregation *TR::DebugCounterGroup::createAggregation(TR::Compilation *comp)
+TR::DebugCounterAggregation *TR::DebugCounterGroup::findAggregation(const char *nameChars, int32_t nameLength)
+   {
+   if (nameChars == NULL)
+      return NULL;
+   char *name = (char*)alloca(nameLength+1);
+   strncpy(name, nameChars, nameLength);
+   name[nameLength] = 0;
+
+   // There's a race here if we do parallel compilation
```

Comment:
Don't you mean there would be a race if we do parallel compilation? The critical region prevents the race, does it not? Or is there another race here you're referring to?

Reply:
I added that line because there's a similar comment here https://github.com/eclipse/omr/pull/2408/files/4f5f0fa137ffcb00f92a01f91654835fc6ff07bb#diff-d89c18db7b5f8e5d8da1080cb3818635R563

I guess the comment describes why there's a critical section